public/app/plugins/datasource/loki/querybuilder/LokiQueryModeller.ts

```diff
@@ -27,8 +27,13 @@ export class LokiQueryModeller extends LokiAndPromQueryModellerBase<LokiVisualQu
 
   renderQuery(query: LokiVisualQuery) {
     let queryString = `${this.renderLabels(query.labels)}`;
-    queryString = this.renderOperations(queryString, query.operations);
-    queryString = this.renderBinaryQueries(queryString, query.binaryQueries);
+    try {
+      queryString = this.renderOperations(queryString, query.operations);
+      queryString = this.renderBinaryQueries(queryString, query.binaryQueries);
+    } catch (err) {
+      // TODO: For now we just log it
+      console.error(err);
```

Comment:
oh ok but this is a different part. I thought we talked about the parsing. Here in rendering erroring out seems a bit weird. like you have an operation in the vis query object but at the same time, you don't know how to render it? In that case, how did it even get into that object? If it was added by the user I would assume we have a definition for it (and so also render code) or if it was parsed from string but we don't have a definition that could be seen as a parsing bug because we also don't know what UI to use for such operation.